activerecord/test/cases/adapters/postgresql/enum_test.rb

```diff
@@ -19,35 +19,28 @@ class PostgresqlEnum < ActiveRecord::Base
     }, prefix: true
   end
 
-  def setup
-    @connection = ActiveRecord::Base.lease_connection
-    @connection.transaction do
-      @connection.create_enum("mood", ["sad", "ok", "happy"])
-      @connection.create_table("postgresql_enums") do |t|
-        t.column :current_mood, :mood
-      end
-    end
-  end
-
-  teardown do
+  def teardown
     reset_connection
     @connection.drop_table "postgresql_enums", if_exists: true
-    @connection.drop_enum "mood", if_exists: true
+    @connection.enum_types.each do |enum_type|
+      @connection.drop_enum enum_type
+    end
     reset_connection
   end
 
   def test_column
+    PostgresqlEnum.reset_column_information
     column = PostgresqlEnum.columns_hash["current_mood"]
     assert_equal :enum, column.type
-    assert_equal "mood", column.sql_type
+    assert_equal @enum_type, column.sql_type
```

Comment:
Instead of using instance variable, use a method, so if subclasses don't define it, it will raise an exception instead of just being `nil` here.

Reply:
I like that, done